kubectl/kubectl.bash

```diff
@@ -8,33 +8,42 @@ function var_usage() {
   CLOUDSDK_COMPUTE_REGION=<cluster region> (regional clusters)
   CLOUDSDK_COMPUTE_ZONE=<cluster zone>
   CLOUDSDK_CONTAINER_CLUSTER=<cluster name>
-
+  KUBERNETES_CONFIG_FILE=path to alternate kubectl config in your workspace
   Optionally, you can specify the kubectl version via KUBECTL_VERSION, taking one of the following values: ${versions[@]}
 EOF
 
   exit 1
 }
 
 kubectl_cmd=kubectl${KUBECTL_VERSION:+.${KUBECTL_VERSION}}
-cluster=${CLOUDSDK_CONTAINER_CLUSTER:-$(gcloud config get-value container/cluster 2> /dev/null)}
-region=${CLOUDSDK_COMPUTE_REGION:-$(gcloud config get-value compute/region 2> /dev/null)}
-zone=${CLOUDSDK_COMPUTE_ZONE:-$(gcloud config get-value compute/zone 2> /dev/null)}
-project=${CLOUDSDK_CORE_PROJECT:-$(gcloud config get-value core/project 2> /dev/null)}
-
-[[ -z "$cluster" ]] && var_usage
-[ ! "$zone" -o "$region" ] && var_usage
-if [[ -n "$KUBECTL_VERSION" ]] && [[ ! " ${versions[*]} "  =~ " ${KUBECTL_VERSION} "  ]]; then
-  echo "Bad KUBECTL_VERSION \"${KUBECTL_VERSION}\". Expected one of ${versions[*]}" >&2
-  exit 1
-fi
 
-if [ -n "$region" ]; then
-  echo "Running: gcloud container clusters get-credentials --project=\"$project\" --region=\"$region\" \"$cluster\""
-  gcloud container clusters get-credentials --project="$project" --region="$region" "$cluster" || exit
-else
-  echo "Running: gcloud container clusters get-credentials --project=\"$project\" --zone=\"$zone\" \"$cluster\""
-  gcloud container clusters get-credentials --project="$project" --zone="$zone" "$cluster" || exit
- fi
+if [ -z "$KUBERNETES_CONFIG_FILE" ]; then
```

Comment:
```suggestion
```
I think we still need to fill it even though we already provided the custom kubeconfig aren't we?

Reply:
the idea here is, that a completely google-independent k8s cluster could be addressed. so, no, we dont need to fill gcloud params in such a case. maybe the case you have on your mind also exists... but i didnt think about such one yet. the proposed change targets remote clusters, independent from gcloud.